### PR TITLE
EA-45,49,50 Added validation for patient update, Modified getConditionHi...

### PR DESCRIPTION
**Pull request by sravanthi17**
...story base on the feedback, also changed getActiveconditions to return in reverse chronological order

### CODE REVIEW

File: condition-list/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java [hunk 1]
```diff
@@ -49,8 +52,46 @@ public Condition save(Condition condition) {
...
+
+        List<Condition> groupedConditions = new ArrayList<Condition>();
+
+        if (CollectionUtils.isEmpty(conditionList)) return new ArrayList<ConditionHistory>();
+
+        ConditionHistory conditionhistory = createConditionHistory(conditionList.get(0));
+        groupedConditions.add(conditionList.get(0));
+        for (int i = 1; i < conditionList.size(); i++) {
+            Condition condition = conditionList.get(i);
+            if (condition.getConcept().equals(conditionhistory.getCondition())) {
```
Comment (rkorytkowski) [inline condition-list/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java]: This assumes conditionList will have conditions ordered by concepts, which I don't think is true. They are ordered by dateCreated. I think what you need here is Map<Concept, ConditionHistory> and Map<String, ConditionHistory> (for non coded conditions) and then you can distribute conditionList based on concept or nonCoded value into proper ConditionHistories.

File: condition-list/src/main/java/org/openmrs/api/db/hibernate/HibernateConditionDAO.java [hunk 2]
```diff
@@ -59,9 +59,7 @@ public Condition getConditionByUuid(String uuid) {
     @Override
     @Transactional(readOnly = true)
     public List<Condition> getConditionHistory(Patient patient) {
-        Query query = sessionFactory.getCurrentSession().createQuery("select con from Condition as con " +
-                " inner join fetch con.concept as c left join c.names as name with name.conceptNameType = 'FULLY_SPECIFIED' " +
-                " where con.patient.patientId = :patientId and con.voided = false  group by c.conceptId, con.conditionId order by name.name asc , con.dateCreated desc");
+        Query query = sessionFactory.getCurrentSession().createQuery("select con from Condition as con inner join fetch con.concept as c where con.patient.patientId = :patientId and con.voided = false  group by c.conceptId, con.conditionId order by con.dateCreated desc");
```
Comment (rkorytkowski) [inline condition-list/src/main/java/org/openmrs/api/db/hibernate/HibernateConditionDAO.java]: Can you explain the purpose of "group by" please? I don't understand why you use it here.

File: condition-list/src/main/java/org/openmrs/api/validator/ConditionValidator.java [hunk 3]
```diff
@@ -121,4 +122,11 @@ private void validateNonCodedCondition(Condition condition, Errors errors) {
             }
         }
     }
+
+    private void validatePatient(Condition condition, Errors errors) {
+        Condition existingCondition = conditionService.getConditionByUuid(condition.getUuid());
```
Comment (rkorytkowski) [inline condition-list/src/main/java/org/openmrs/api/validator/ConditionValidator.java]: Can you please instead modify Condition.setPatient to throw an exception if you try to change the value? Please see my last comment on https://issues.openmrs.org/browse/EA-45

**Comment by rkorytkowski** [comment]
Thanks @sravanthi17 !

**Comment by sravanthi17** [comment]
Welcome.. :)

On Fri, Mar 20, 2015 at 6:33 PM, Rafal Korytkowski <notifications@github.com
> wrote:

> Thanks @sravanthi17 <https://github.com/sravanthi17> !
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmrs/openmrs-module-emrapi/pull/44#issuecomment-84007172>
> .
>



-- 
*N. SRAVANTHI,*
*Application Developer,*
*ThoughtWorks,*

**Pull request merged**